src/main/java/com/google/api/codegen/transformer/TestCaseTransformer.java

```diff
@@ -258,29 +262,37 @@ private MockGrpcResponseView createMockResponseView(
 
   private InitCodeContext createResponseInitCodeContext(
       MethodContext context, SymbolTable symbolTable) {
-    ArrayList<FieldModel> primitiveFields = new ArrayList<>();
     TypeModel outputType = context.getMethodModel().getOutputType();
     if (context.getMethodConfig().isLongRunningOperation()) {
       outputType = context.getMethodConfig().getLongRunningConfig().getReturnType();
     }
-    for (FieldModel field : outputType.getFields()) {
-      if (field.getType().isPrimitive() && !field.getType().isRepeated()) {
-        primitiveFields.add(field);
-      }
-    }
     return InitCodeContext.newBuilder()
         .initObjectType(outputType)
         .symbolTable(symbolTable)
         .suggestedName(Name.from("expected_response"))
         .initFieldConfigStrings(ImmutableList.<String>of())
         .initValueConfigMap(ImmutableMap.<String, InitValueConfig>of())
-        .initFields(primitiveFields)
+        .initFields(responseInitFields(outputType.getFields()))
         .fieldConfigMap(context.getProductConfig().getDefaultResourceNameFieldConfigMap())
         .valueGenerator(valueGenerator)
         .additionalInitCodeNodes(createMockResponseAdditionalSubTrees(context))
         .build();
   }
 
+  @VisibleForTesting
+  static <E extends FieldModel> List<FieldModel> responseInitFields(List<E> fields) {
+    HashSet<Oneof> oneofSet = new HashSet<>();
+    return fields
+        .stream()
+        .filter(f -> f.isPrimitive() && !f.isRepeated())
+        .filter(
+            f -> {
+              Oneof oneof = f.getOneof();
+              return oneof == null || oneofSet.add(oneof);
```

Comment:
Could you add a comment stating that this includes a field if it either isn't a oneof, or it's the first field on a oneof?